mlir/lib/Dialect/LLVMIR/IR/LLVMLinkerInterface.cpp

```diff
@@ -283,6 +320,22 @@ class LLVMSymbolLinkerInterface : public SymbolLinkerInterface {
     setVisibility(pair.src, visibility);
     setVisibility(pair.dst, visibility);
 
+    UnnamedAddr srcUnnamedAddr = getUnnamedAddr(pair.src);
+    UnnamedAddr dstUnnamedAddr = getUnnamedAddr(pair.dst);
+
+    if (isAppendingLinkage(srcLinkage) && isAppendingLinkage(dstLinkage)) {
+      if (srcUnnamedAddr != dstUnnamedAddr) {
+        pair.src->emitError("Appending variables with different unnamed_addr "
+                            "need to be linked: ")
+            << pair.dst->getLoc();
+        return failure();
+      }
+    }
```

Comment:
Where did this come from?

Reply:
In llvm-link it happens in `lib/Lnker/IRMover.cpp:877`. I added it to satisfy the imported test. Should I move it somewhere else?

Reply:
Yes, this is to verbose and disctracts from logic.

Add some helper lambda `linkError` with use here like 
```
return linkError("Appending variables with different unnamed_addr need to be linked:");
```

Reply:
And ideally extract entire thing to separate check function, because there will be more conditions on src and dst here. E.g. result should look like:
```
 if (isAppendingLinkage(srcLinkage) && isAppendingLinkage(dstLinkage)) {
  if (failed(checkAppendedVars(src, dst)))
    return failure();
}
```